packages/twenty-server/src/engine/core-modules/auth/services/auth.service.ts

```diff
@@ -230,7 +284,28 @@ export class AuthService {
     }
 
     if (isDefined(workspace)) {
-      workspaceValidator.isAuthEnabledOrThrow(authParams.provider, workspace);
+      const isProviderEnabled = workspaceValidator.isAuthEnabled(
+        authParams.provider,
+        workspace,
+      );
+
+      if (!isProviderEnabled) {
```

Comment:
You should instead do an early return before if provider is enabled? Here you'll do one db query for each login while it's not needed in 99% of cases because they didn't go through the bypass form